packages/server/src/fhircast/websocket.ts

```diff
@@ -17,10 +73,34 @@ export async function handleFhircastConnection(socket: ws.WebSocket, request: In
   // According to Redis documentation: http://redis.io/commands/subscribe
   // Once the client enters the subscribed state it is not supposed to issue any other commands,
   // except for additional SUBSCRIBE, PSUBSCRIBE, UNSUBSCRIBE and PUNSUBSCRIBE commands.
-  const redisSubscriber = getRedis().duplicate();
+  const redis = getRedis();
+  const redisSubscriber = redis.duplicate();
+  const deferredCleanupPromise = createDeferredPromise();
+  cleanupPromises.push(deferredCleanupPromise.promise);
```

Comment:
It looks like we're only adding to `cleanupPromises`, and never removing?  If lots of frequent connect/disconnects, would this array continue to grow?  Probably fine in the short term when not many users, but ideally we'd remove promises from the array on websocket close.

Reply:
Good point, I think in an earlier iteration I couldn't figure out how to safely remove promises reliably, but I guess when they are resolved they can probably just be removed. I'll fix that